docs/linux/sched.md

```diff
@@ -1,167 +1,173 @@
-SCSI disk driver, below the block layer, operating on requests already passed to
-the device dispatch queue by the block I/O scheduler.
+Zone write locking was first implemented in kernel 4.10 in the SCSI disk
+driver (below the block layer), operating on requests already passed to the
+device dispatch queue by the block I/O scheduler.
 
-This early implementation was relying on the fact that the SCSI layer may delay
+This early implementation relied on the fact that the SCSI layer could delay
 issuing any request to the device. By maintaining a bitmap with one bit per
-zone, the SCSI disk driver marked a zone as *locked* whenever a write command
-was seen. In more details, the algorithm is as follows.
+zone, the SCSI disk driver marked a zone as *locked* whenever it saw a write
+command. This algorithm is presented here in more detail:
 
-1. If the next command to dispatch to the device is not a write command, the
-   command is dispatched immediately.
+1. If the next command to be dispatched to the device is not a write command,
+   then the command is dispatched immediately.
 
-2. If the next command to dispatch is a write command, the zone write lock bit
-   for the target zone of the command is inspected.
+2. If the next command to by dispatched is a write command, then the zone
+   write lock bit for the target zone of the command is inspected.
 
-    1. If the target zone of the write command is not write locked (bit not
-       set), then the zone is locked and the write command issued to the device.
-       Both operations are atomically executed under the device request queue
-       spinlock.
+    1. If the target zone of the write command is not write locked (i.e., the
+       bit is not set), then the zone is locked and the write command issued
+       to the device.  Both operations are atomically executed under the
+       device request queue spinlock.
 
-    2. If the target zone is already locked (bit set), the SCSI disk driver
-       temporarily delays the command issuing to the device until the zone write
-       lock is released.
+    2. If the target zone is already locked (i.e., the bit is set), then the
+       SCSI disk driver temporarily delays issuing the command to the device
+       until the zone write lock is released.
 
-3. When a write command completes, the zone write lock for the target zone of
-   the command is released and the dispatch process resumed. This means that if
-   the command at the head of the dispatch queue is targeting the same zone,
-   this command can now be issued (step 2.a).
+    3. When a write command completes, the zone write lock for the target zone
+       of the command is released and the dispatch process is resumed. This
+       means that if the command at the head of the dispatch queue targets the
+       same zone, then the command can be issued (when the write command
+       completes) (step 2.a).
```

Comment:
Nice rewrite. But the sub-clause of step (2) are numbered 1, 2 and 3, not a, b and c... So referencing 2.a is not working. Can you change the sub-bullets to be a, b and c ?

Reply:
Good catch.

I've made this alteration.